_posts/2025-04-28-running-at-scale.adoc

```diff
@@ -0,0 +1,143 @@
+Additionally, it can help reduce the memory footprint of OLM, as it no longer needs to maintain and manage these additional resources.
+
+TIP: More details about Disabling Copied CSVs can be found in the link:https://olm.operatorframework.io/docs/advanced-tasks/configuring-olm/#disabling-copied-csvs[official documentation].
+
+== Cluster Autoscaling
+
+Although cluster autoscaling is a powerful Kubernetes feature, you cannot always fall back on it and should always consider predictive scaling by analyzing load data on your environment to detect daily or weekly usage patterns. If your workloads follow a pattern and there are dramatic peaks throughout the day, you should consider provisioning worker nodes accordingly. For example, if you have a predictable load pattern where the number of workspaces increases during business hours and decreases during off-hours, you can use predictive scaling to adjust the number of worker nodes based on the expected load.
+This can help ensure that you have enough resources available to handle the peak load while minimizing costs during off-peak hours.
+
+== Multicluster
+
+By design, Eclipse Che is not multicluster aware, and you can only have one instance per cluster. However, it is possible to run Eclipse Che in a multicluster environment by deploying Eclipse Che in different clusters and using a load balancer or DNS-based routing to direct traffic to the appropriate instance based on the user’s location or other criteria. This approach can help improve performance and reliability by distributing the workload across multiple clusters and providing redundancy in case of cluster failures.
+
+One example of a multicluster architecture is link:https://workspaces.openshift.com/[workspaces.openshift.com], which is part of the link:https://developers.redhat.com/developer-sandbox[Red Hat Developer Sandbox].
+
+From the infrastructure perspective, the Red Hat Developer Sandbox consists of multiple link:https://www.redhat.com/en/technologies/cloud-computing/openshift/aws[ROSA] clusters. On each cluster, the productized version of Eclipse Che is installed and configured using link:https://argo-cd.readthedocs.io/en/stable/[Argo CD]. Since the user base is spread across multiple clusters, link:https://workspaces.openshift.com/[workspaces.openshift.com] is used as a single entry point to the productized Eclipse Che instances. You can find implementation details about the multicluster redirector in the following link:https://github.com/codeready-toolchain/crw-multicluster-redirector[GitHub repository].
+
+TIP: link:https://workspaces.openshift.com/[workspaces.openshift.com] is a Developer Sandbox-specific solution that can not be reused as-is in other environments. However, it can be used as a reference for implementing a similar solution well-tailored for your specific multicluster needs.
```

Comment:
```suggestion
TIP: The solution for link:https://workspaces.openshift.com/[workspaces.openshift.com] is a Developer Sandbox-specific solution that can not be reused as-is in other environments. However, it can be used as a reference for implementing a similar solution well-tailored for your specific multicluster needs.
```